searches/b.py

```diff
@@ -1,143 +1,216 @@
+                        self.non_conflicts[point] = [chosen_square]
+                        stack_remove.append(point)
+                elif len(point.squares) == 1:
+                    if point.squares[0] in self.conflicts:
+                        for square in self.conflicts[point.squares[0]]:
+                            candidat_remove.append(square)
+                        if point not in self.non_conflicts:
+                            self.non_conflicts[point] = []
+                        self.non_conflicts[point].append(point.squares[0])
+                else:
+                    for square in point.squares:
+                        if square in self.conflicts:
+                            for sq1, sq2 in itertools.combinations(self.conflicts[square], 2):
                                 if sq1.point == sq2.point and len(sq1.point.squares) == 2:
-                                    B.remove_candidat(square, conflicts)
-                                    # print(f"overlaps 2 finals: {square.point} , {square.orientation}")
+                                    candidat_remove.append(square)
                                     break
-            # if pop:
-            #     stack.pop()
-
-        for point in self.points:
-            if len(point.squares) == 0:
+                if candidat_remove:
+                    for candidat in candidat_remove:
+                        self.remove_candidat(candidat)
+            if stack_remove:
+                for remove in stack_remove:
+                    stack.remove(remove)
+        if not self.posible:
+            return False
+        return True
+
+    def overlap(self, x, y):
+        for key, value in x.items():
+            x_sq = key
+            x_t = value
+        for key, value in y.items():
+            y_sq = key
+            y_t = value
+        if x_t and y_t:
+            if y_sq in self.conflicts[x_sq]:
                 return False
-            if len(point.squares) > 1:
-                hold = point.squares[0]
-                for square in point.squares:   
-                    B.remove_candidat(square, conflicts)
-                    # print(f"after: {point} , {square.orientation}")
-                point.squares.append(hold)
-                # print(f"hold : {point} , {hold.orientation}")
-
-        square_orientation = [point.squares[0] for point in self.points]
-
-        return square_orientation
+        return True
+
+    def count_true(solution):
+        if solution:
+            values = []
+            for value in solution.values():
+                if value.values():
+                    tmp = list(value.values())
+                    values = values + tmp 
+            return sum(1 for value in values if value)
+        else:
+            return 0
+
+    def solution_to_dict(solution):
+        if solution:
+            keys = []
+            values = []
+            for key in solution.values():
```

Comment:
`reduce()` pa se snadji

Reply:
?

Reply:
Ima funkcija `reduce` u pajtonu koja agregira sumu neke iterable po nekoj varijabli. Guglaj pa vidi dal moze da se iskotisti ovde.

Reply:
mislim da je to izbaceno